codecov.yml

```diff
@@ -23,15 +23,68 @@ coverage:
       default:
         branches:
           - develop
+        target: 90%
         informational: true
         only_pulls: true
         paths:
-          - "nncf/onnx"
-          - "nncf/common"  # extend this once we collect coverage reports for more than just onnx and common part of precommit
+          - "nncf"
 
 comment:
-  layout: "diff, flags, files"
+  layout: "reach, diff, files, flags, components"
   require_changes: false
 
   require_head: false
   require_base: false
+
+flag_management:
+  # Flag coverage percentage seems to show the "percentage of lines under the flag path covered as reported ONLY
+  # by the upload with the corresponding flag", so e.g. for COMMON the flag coverage percentage will report the
+  # percentage of common code tested ONLY by the common tests, and e.g. not by backend-specific precommit parts
+  # (which also run common code and are therefore indirectly providing coverage). Ideally each flag-specific path
+  # would be described below with the corresponding flag and provide valuable information on whether the test code base
+  # is written efficiently, e.g. that the backend-specific tests predominantly validate backend-specific code and the
+  # common tests completely cover the common code on their own. However, if we set all flags with paths here, then the
+  # total repo coverage percentage will sink, because codecov currently reports the overall coverage based on the union
+  # of the "flag" coverages - not the "component" coverages (see below) - and currently NNCF's precommit tests are
+  # biased toward validating common code via backend-specific tests. In the future the tests will be gradually
+  # refactored to have more "locality" in what each precommit section tests.
+  individual_flags:
+    - name: COMMON
+      paths:
+        - nncf/common
+        - nncf/quantization
+
+component_management:
+  # In contrast to the "flag" coverage above, the "component" display seems to calculate percentage based on the
+  # coverage information from ALL uploads for the code in the specified path. With this, the "component" coverage
+  # percentage is a better representation of what sub-paths in the NNCF code base are covered with at least one test,
+  # without distinction whether the test was run in the
+  individual_components:
+    - component_id: common
+      name: common
+      paths:
+        - nncf/common
+    - component_id: torch
+      name: torch
+      paths:
+        - nncf/torch
+        - nncf/**/torch_*.py
+    - component_id: tensorflow
+      name: tensorflow
+      paths:
+        - nncf/tensorflow
+        - nncf/**/tensorflow_*.py
+    - component_id: onnx
```

Comment:
![image](https://github.com/openvinotoolkit/nncf/assets/48012821/e1092857-9a12-4032-b5f3-ee1fa7040864)
![image](https://github.com/openvinotoolkit/nncf/assets/48012821/f747c9e2-6c9f-44f5-9ef3-b06790cc0f1c)

is it expected?

Reply:
The corresponding precommit stage did not submit a coverage report, it seems; looks like a spurious CI failure.
https://app.codecov.io/gh/openvinotoolkit/nncf/commit/66ad33290d7796d40261c09b2315111d2d35dbb1

Reply:
Codecov uploader stage runs into a 502 error sporadically, it seems:
```
16:36:12  [2023-11-22T15:36:10.687Z] ['error'] There was an error running the uploader: Error uploading to https://codecov.io/: Error: There was an error fetching the storage URL during POST: 502 - 
16:36:12  <html><head>
16:36:12  <meta http-equiv="content-type" content="text/html;charset=utf-8">
16:36:12  <title>502 Server Error</title>
16:36:12  </head>
16:36:12  <body text=#000000 bgcolor=#ffffff>
16:36:12  <h1>Error: Server Error</h1>
16:36:12  <h2>The server encountered a temporary error and could not complete your request.<p>Please try again in 30 seconds.</h2>
16:36:12  <h2></h2>
16:36:12  </body></html>
```